lib/Pipes/TaggedFunctionKind.cpp

```diff
@@ -50,6 +52,107 @@ using TaggedFK = TaggedFunctionKind;
 void TaggedFK::getInvalidations(const Context &Ctx,
                                 TargetsList &ToRemove,
                                 const GlobalTupleTreeDiff &Diff) const {
+  return;
+}
+
+using IsolatedFK = IsolatedFunctionKind;
+void IsolatedFK::getInvalidations(const Context &Ctx,
+                                  TargetsList &ToRemove,
+                                  const GlobalTupleTreeDiff &Diff) const {
+  using PM = PathMatcher;
+  using Binary = model::Binary;
+
+  const auto &CurrentModel = getModelFromContext(Ctx);
+  if (not Ctx.containsReadOnlyContainer(BinaryCrossRelationsRole))
+    return;
```

Comment:
The `BinaryCrossRelationsRole` is only for refining the set of things you need to remove.
If you don't have it, you must invalidate *all* the targets. Here you're invalidating no target at all.

Reply:
The [upcoming structure](https://github.com/revng/revng/pull/209/commits/33268127353b18637111e97de73726f5c3c20988) of the Pipes guarantees the `BinaryCrossRelationsRole` is created if there at least a function has been isolated. If `CrossRelations` container does not exist, there are no isolated functions at all, thus returning should be reasonable.

Reply:
It should be OK now, we're invalidating all the targets (by specifying only the rank), sounds good?

Reply:
See the other comment